src/main/java/nwoolcan/model/brewery/production/batch/review/BatchEvaluationImpl.java

```diff
@@ -0,0 +1,65 @@
+package nwoolcan.model.brewery.production.batch.review;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Implementation of {@link BatchEvaluation}.
+ */
+public final class BatchEvaluationImpl extends EvaluationImpl implements BatchEvaluation {
+    private final Collection<Evaluation> categories;
+    private final Optional<String> reviewer;
+
+    BatchEvaluationImpl(final BatchEvaluationType type, final Collection<Evaluation> categories, final Optional<String> reviewer, final Optional<String> notes) {
```

Comment:
should not pass optional as argument, but check for null instead

Reply:
I do not like this option, passing Optional<> make explicit that the argument could be absent and that the implementation handles both cases

Reply:
this is because you are using `Optional` even in fields
in java it is highly recommended not to use `Optional` for field types or argument types, but if you want to use them like this is your chioce.
i prefer to stand by the java guidelines.
[this](https://blog.joda.org/2014/11/optional-in-java-se-8.html) is a post explaining it

Reply:
> I do not like this option, passing Optional<> make explicit that the argument could be absent and that the implementation handles both cases

That has nothing to do with using `Optional` even in fields. If I'll need to remove `Optional` field to make the class serializable I will, but not now when serializing methods are not defined yet